tools/generators/move-packages/index.ts

```diff
@@ -0,0 +1,131 @@
+import { Tree, formatFiles, joinPathFragments, updateJson, readJson, ProjectConfiguration } from '@nrwl/devkit';
+import { moveGenerator } from '@nrwl/workspace/generators';
+import { PackageJson } from '../../types';
+import { getProjects } from '../../utils';
+
+import { MovePackagesGeneratorSchema } from './schema';
+
+export default async function (tree: Tree, schema: MovePackagesGeneratorSchema) {
+  validateSchema(schema);
+
+  if (hasSchemaFlag(schema, 'allConverged')) {
+    runBatchMove(tree, schema, isPackageConverged);
+  } else if (hasSchemaFlag(schema, 'allV8')) {
+    runBatchMove(tree, schema, isV8Package);
+  } else {
+    movePackage(tree, schema);
+  }
+
+  await formatFiles(tree);
+}
+
+function runBatchMove(
+  tree: Tree,
+  schema: MovePackagesGeneratorSchema,
+  libraryVersionChecker: (tree: Tree, project: ProjectConfiguration) => boolean,
+) {
+  const projects = getProjects(tree);
+  for (const [projectName, projectConfig] of projects) {
+    if (libraryVersionChecker(tree, projectConfig)) {
+      const destination = `${schema.destination}/${projectName.split('/')[1]}`;
+      console.log(`Attempting to move ${projectName} to ${destination}`);
+
+      movePackage(tree, {
+        name: projectName,
+        destination: destination,
+        updateImportPath: schema.updateImportPath,
+      });
+    }
+  }
+}
+
+function movePackage(tree: Tree, schema: MovePackagesGeneratorSchema) {
+  const { name, destination, updateImportPath = false } = schema;
+
+  if (!name) {
+    return;
+  }
+
+  moveGenerator(tree, {
+    projectName: name,
+    destination: destination,
+    importPath: name,
+    updateImportPath: updateImportPath,
+  });
+
+  // moveGenerator automatically renames the package so this overwrites that change
```

Comment:
mind adding an example of how the moveGenerator will rename the package ?

Reply:
Yup so lets say you want to move `react-menu` to a new subfolder called testFolder/react-menu using `yarn nx workspace-generator move-packages --name='@fluentui/react-menu' --destination="testFolder/react-menu"` - the `moveGenerator` will delete `@fluentui/react-menu` from the `workspace.json` and add a new entry with the name of `testFolder-react-menu`. This auto renaming is also why I'm setting `importPath` to be the `name` of the package being moved along with defaulting `updateImport` to `false` to prevent further renames in various packages that reference the moved package.